internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -80,36 +81,110 @@ func getCSVInstallStrategy(csv *operatorsv1alpha1.ClusterServiceVersion) operato
-	perms := []operatorsv1alpha1.StrategyDeploymentPermissions{}
-	for _, role := range c.ClusterRoles {
-		perms = append(perms, operatorsv1alpha1.StrategyDeploymentPermissions{
-			ServiceAccountName: role.GetName(),
-			Rules:              role.Rules,
-		})
+	// Collect all cluster role service accounts by their corresponding service accounts via bindings.
+	// This lets us look up all service accounts a cluster role is bound to and create one set of clusterPermissions
+	// per service account.
+	saNamesToClusterRoleNames := make(map[string]map[string]struct{})
+	for _, binding := range c.ClusterRoleBindings {
+		roleRef := binding.RoleRef
+		if roleRef.Kind == "ClusterRole" && (roleRef.APIGroup == "" || roleRef.APIGroup == rbacv1.SchemeGroupVersion.Group) {
+			for _, name := range getSubjectServiceAccountNames(binding.Subjects) {
+				if saNamesToClusterRoleNames[name] == nil {
+					saNamesToClusterRoleNames[name] = make(map[string]struct{})
+				}
+				saNamesToClusterRoleNames[name][roleRef.Name] = struct{}{}
+			}
+		}
+	}
+
+	// Apply relevant cluster roles to each service account.
+	unmappedClusterRoles := []rbacv1.ClusterRole{}
+	clusterPerms := []operatorsv1alpha1.StrategyDeploymentPermissions{}
+	for saName, clusterRoleNames := range saNamesToClusterRoleNames {
+		p := operatorsv1alpha1.StrategyDeploymentPermissions{ServiceAccountName: saName}
+		for _, clusterRole := range c.ClusterRoles {
+			if _, ok := clusterRoleNames[clusterRole.GetName()]; ok {
+				p.Rules = append(p.Rules, clusterRole.Rules...)
+			} else {
+				unmappedClusterRoles = append(unmappedClusterRoles, clusterRole)
+			}
+		}
+		clusterPerms = append(clusterPerms, p)
+	}
+	strategy.ClusterPermissions = clusterPerms
+	// Add all unmapped cluster roles to a clusterPermission without a service account.
+	if len(unmappedClusterRoles) != 0 {
+		unmappedClusterPerm := operatorsv1alpha1.StrategyDeploymentPermissions{}
+		for _, role := range unmappedClusterRoles {
+			unmappedClusterPerm.Rules = append(unmappedClusterPerm.Rules, role.Rules...)
+		}
+		strategy.ClusterPermissions = append(strategy.ClusterPermissions, unmappedClusterPerm)
+	}
+}
+
+// getSubjectServiceAccountNames returns a list of all ServiceAccount subject names.
+func getSubjectServiceAccountNames(subjects []rbacv1.Subject) (saNames []string) {
+	for _, subject := range subjects {
+		if subject.Kind == "ServiceAccount" {
+			saNames = append(saNames, subject.Name)
+		}
 	}
-	strategy.ClusterPermissions = perms
+	return saNames
 }
 
 // applyDeployments updates strategy's deployments with the Deployments
 // in the collector.
 func applyDeployments(c *collector.Manifests, strategy *operatorsv1alpha1.StrategyDetailsDeployment) {
 	depSpecs := []operatorsv1alpha1.StrategyDeploymentSpec{}
 	for _, dep := range c.Deployments {
+		// TODO(estroz): ensure each deployment's pod template spec contains a service account. There may be more
```

Comment:
The CSV will install successfully without one (tested using an example operator) so this isn't necessary to deal with right away.

Reply:
I think we just use what's in the deployment. If there isn't one listed, that implicitly means `default`.